drivers/memc/Kconfig.stm32

```diff
@@ -1,9 +1,12 @@
 # Copyright (c) 2020 Teslabs Engineering S.L.
 # SPDX-License-Identifier: Apache-2.0
 
+DT_COMPAT_ST_STM32_FMC := st,stm32-fmc
+
 config MEMC_STM32
 	bool "Enable STM32 Flexible Memory Controller (FMC)"
-	depends on SOC_SERIES_STM32H7X
+	depends on $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FMC))
+	default y
```

Comment:
`default $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FMC))`

Reply:
So I'm dropping the `depends` and using the compat for `default`. Fine by me, though then it allows selecting the option without the node being present.

Reply:
...it also allows selecting the driver on any architecture, let me know if I misinterpreted the suggested change.

Reply:
I think that Kconfig needs to specify software dependencies, that is, the driver implementation depends on H7 and F4 series not on a node enabled in DT. With current solution, one can also define the FMC node in any series and get it compiled. So I guess the right way would be:

```
depends on SOC_SERIES_STM32H7X || SOC_SERIES_STM32F4X
default $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FMC))
```

Reply:
Ok changing it to that. Thing is that those lists are getting out of control for some of the other drivers. Thinking that enabling these from the soc with a series of `HAS_STM32_XXX` options may be cleaner. Anyway, not in scope for this PR so let's go with your suggestion.

Reply:
looks like e49c849 commit message needs to be adjusted now

Reply:
Right, updated.